src/bootstrap/test.rs

```diff
@@ -911,7 +917,20 @@ impl Step for Compiletest {
             cmd.arg("--lldb-python-dir").arg(dir);
         }
 
-        cmd.args(&build.config.cmd.test_args());
+        // Get paths from cmd args
+        let paths = match build.config.cmd {
+            Subcommand::Test { ref paths, ..} => &paths[..],
+            _ => &[]
+        };
+
+        // Get test-args by striping suite path
+        let mut test_args: Vec<&str> = paths.iter().filter(|p| p.starts_with(suite_path) &&
+           p.is_file()).map(|p| p.strip_prefix(suite_path).unwrap().to_str().unwrap()).collect();
```

Comment:
We should probably error out here if the argument is going to be silently ignored because it's part of our suite yet not a file (and as such we don't know what to do with it).